modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -421,90 +421,41 @@ public PathElement[] getCaretShape(int offset, boolean isLeading,
     }
 
     @Override
-    public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRunStart) {
+    public Hit getHitInfo(float x, float y) {
         int charIndex = -1;
         int insertionIndex = -1;
         boolean leading = false;
-        int relIndex = 0;
-        int textWidthPrevLine = 0;
 
         ensureLayout();
-        int lineIndex = getLineIndex(y, text, curRunStart);
+        int lineIndex = getLineIndex(y);
         if (lineIndex >= getLineCount()) {
             charIndex = getCharCount();
             insertionIndex = charIndex + 1;
         } else {
-            if (isMirrored()) {
-                x = getMirroringWidth() - x;
-            }
             TextLine line = lines[lineIndex];
             TextRun[] runs = line.getRuns();
             RectBounds bounds = line.getBounds();
             TextRun run = null;
             x -= bounds.getMinX();
             //TODO binary search
```

Comment:
we can remove this TODO because the new approach is incompatible with the binary search